### ISSUE TITLE
Documentation needed on System.Collections.Immutable

### ISSUE THREAD

**Issue by bluemmc** [dotnet/corefx#5165]
The documentation for how to use System.Collections.Immutable is the worst I have seen of any general microsoft libraries. No general introduction and very few examples. + Most stuff that google can find on the net is outdated and may no longer work. I need quite some detective work just to find out how to instantiate an instance of an immutable list with specific values, how to convert a IEnumerable etc.

**Comment by stephentoub** [comment]
cc: @rpetrusha, @AArnott

**Comment by AArnott** [comment]
Team: Where would such documentation go? MSDN? xml doc comments aren't going to make examples discoverable.

**Comment by bluemmc** [comment]
Great. Thanks for taking my input into consideration. I will be happy to give feedback on new documentation proposals if you are interested.

**Comment by bluemmc** [comment]
Tip: There is some related old BCL blog entries with examples where some work partly and some does not at all because the API has since changed. Maybe they could be used as a starting point when writing documentation.

**Comment by bluemmc** [comment]
@AArnott Any progress?

**Comment by eriawan** [comment]
@AArnott , @stephentoub 

Any progress about this issue? Can we have simply have an Open Publishing like Visual F# documentation on Github?

For references: https://github.com/microsoft/visualfsharpdocs/

**Comment by stephentoub** [comment]
@mairaw, @cartermp, could you share here on status / plans around docs? Thanks.

**Comment by eriawan** [comment]
@stephentoub 

Just FYI, maybe you have known, but I have just found out that .NET Core has quite the same open publishing like Visual F# docs.
The live site is at: https://docs.microsoft.com/en-us/dotnet/articles/welcome
The github repo is at: https://github.com/dotnet/core-docs

I may miss something here, but I wonder, is there a .NET CoreFX documentation site available for us to look at?

**Comment by stephentoub** [comment]
> maybe you have known

Yup, that's why I cc'd the folks I did ;) They should be able to share more info.

**Comment by eriawan** [comment]
Great!

Thanks @stephentoub :+1:

**Comment by cartermp** [comment]
I created an issue to track and discuss how this content would look in the core-docs repo, where the content would ultimately live.  Since this is conceptual content, this would definitely go there.  As for where in the doc set it ultimately lives, that would be up for discussion in the issue.

**Comment by mairaw** [comment]
Improving the Immutable Collections docs has been on my backlog for a while. Now that it's part of the product with .NET Core, hopefully we can make the case to prioritize this work and make it happen.

Now, in terms of status, just to give you a more complete picture of where we are at the moment. The conceptual docs for .NET Core is now open source just like the F# docs. However, we're still working towards enabling the same experience for the API ref. You can see this issue on our repo that discusses our proposal for the API experience: https://github.com/dotnet/core-docs/issues/772
